### PR TITLE
Split Modules / Stats / One Hot / SVAEC

### PR DESCRIPTION
**Pull request by Edouard360**
* implementation of M1+M2 (see #59), with name SVAEC (Stacked VAE for Classification)
* create a Stats object to store statistics
* Organize scvi package
	* create metrics folder recording (stats object moved in metrics)
	* move modules in models
	* isolate utilities in different files
* Simplify modules Decoder / Encoder / EncoderSCVI by factorizing layers creation
* Creating OneHot module flexible with arguments
* train function **now iterates over both train and test dataloaders**, for semi-supervised training.
* decorators: in utils.py, for torch.no_grad and torch.enable_grad and .train .eval. Code is cleaner
* normalize some attribute names using_cuda -> use_cuda (as flag in run_benchmarks), tensor_list -> tensors (shorter)
* .cuda() called after module __init__ according to use_cuda() arg.
* Use torch.distributions.kl.kl_divergence for kl computation
* Minor change to retina (needed for the unit test labels to be contiguous and for the batch size to be bigger (?))

### CODE REVIEW

**Comment by jeff-regier** [comment]
Looks good. You may want to start using a validation set (i.e. `validation_loader`) rather than using the test set for training.

**Comment by jeff-regier** [comment]
I'm eager to hear about the results. The semi-supervised ladder network might be a good baseline for comparison:

https://github.com/jibancanyang/Semi-supervised_Neural_Network

It doesn't model the data with a ZINB distribution, so we may be able to come out ahead. On the other hand, it has some very impressive results on MNIST.

File: scvi/benchmark.py [hunk 1]
```diff
@@ -26,54 +24,144 @@ def run_benchmarks(gene_dataset, n_epochs=1000, learning_rate=1e-3, use_batches=
     # - batch mixing
     # - cluster scores
     example_indices = np.random.permutation(len(gene_dataset))
-    tt_split = int(0.9 * len(gene_dataset))  # 10%/90% test/train split
+    tt_split = int(0.5 * len(gene_dataset))  # 50%/50% train/test split
```
Comment (jeff-regier) [inline scvi/benchmark.py]: Probably we want to pass the proportion of data to use for testing as an argument to run benchmarks, and default to 10%.

**Comment by Edouard360** [comment]
I open #87 to discuss about comparing with other baselines.

On Chenlin synthetic dataset I can get around to 80% validation accuracy on test set (without much cross validation on the hyperparameters).

Using "naively" the repo you mentioned however, I get only 30% accuracy, so either the ZINB modelling is crucial to the model, or there are hyperparameters that need to be investigated...
